### PR TITLE
Added Gauss-Seidel Method

### PR DESCRIPTION
**Pull request by MUCCHU**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: linear_algebra/src/gauss_seidel.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        a: coefficient matrix
+        x: solution vector
+        b: constant vector
+
+        Example:
+        >>> a = [[10, -1, 2, 0], [-1, 11, -1, 3], [2, -1, 10, -1], [0, 3, -1, 8]]
+        >>> x = [0, 0, 0, 0]
+        >>> b = [6, 25, -11, 15]
+        >>> update_solution(a, x, b)
+        [0.6, 2.3272727272727276, -0.9872727272727271, 0.8788636363636363]
```
Comment (CaedenPH) [inline linear_algebra/src/gauss_seidel.py:24]:
```suggestion
    Function to update the solution
    a: coefficient matrix
    x: solution vector
    b: constant vector

    Example:
    >>> a = [[10, -1, 2, 0], [-1, 11, -1, 3], [2, -1, 10, -1], [0, 3, -1, 8]]
    >>> x = [0, 0, 0, 0]
    >>> b = [6, 25, -11, 15]
    >>> update_solution(a, x, b)
    [0.6, 2.3272727272727276, -0.9872727272727271, 0.8788636363636363]
```

**Review by CaedenPH — COMMENTED** [review]
You should unindent the docstrings

**Comment by MUCCHU** [comment]
> 

Was it the reason for tests failing?

**Comment by MUCCHU** [comment]
pre-commit.ci run

**Pull request closed without merging**